src/main/java/com/battleship/backend/models/ComputerPlayer.java

```diff
@@ -0,0 +1,117 @@
+    Ship[] shipList;
+    Boardable playerBoard;
+    Boardable computerBoard;
+    String name;
+    Validator hitRequestValidator;
+    Validator placeShipsValidator;
+    RequestFactory requestFactory;
+    Random rand;
+
+    public ComputerPlayer(BoardRepository boardRepository, Ship[] shipList, Validator placeShipsValidator, Validator hitRequestValidator){
+        this.boardRepository = boardRepository;
+        this.shipList = shipList;
+        this.playerBoard = boardRepository.getPlayerBoard();
+        this.computerBoard = boardRepository.getComputerBoard();
+        this.hitRequestValidator = hitRequestValidator;
+        this.placeShipsValidator = placeShipsValidator;
+
+        rand = new Random();
+        requestFactory = new RequestFactory();
+        name = "Computer Player";
+    }
+
+    public Request generateValidHitRequest(){
+        int randRow = rand.nextInt(playerBoard.getSize());
+        int randCol = rand.nextInt(playerBoard.getSize());
+
+        Request randHitRequest = requestFactory.generateHitRequest(randRow, randCol);
+
+        while(!hitRequestValidator.isValid(playerBoard, randHitRequest)){
+            randRow = rand.nextInt(playerBoard.getSize());
+            randCol = rand.nextInt(playerBoard.getSize());
+
+            randHitRequest = requestFactory.generateHitRequest(randRow, randCol);
+        }
+        return randHitRequest;
+    }
+
+    public Request generateValidPlaceRequest(Ship ship){
+        int randRow = rand.nextInt(computerBoard.getSize());
+        int randCol = rand.nextInt(computerBoard.getSize());
+
+        Request randPlaceRequest = requestFactory.generatePlaceRequest(ship, randRow, randCol);
+
+        while(!placeShipsValidator.isValid(computerBoard, randPlaceRequest)){
+            randRow = rand.nextInt(computerBoard.getSize());
+            randCol = rand.nextInt(computerBoard.getSize());
+
+            randPlaceRequest = requestFactory.generatePlaceRequest(ship, randRow, randCol);
+        }
+        return randPlaceRequest;
+    }
+
+    public void placeShip(Request placeRequest){
+        computerBoard.addShip(placeRequest.getShip(), placeRequest.getRow(), placeRequest.getCol());
+        removeShip(placeRequest.getShip());
+    }
+
+    public void placeAllShips(){
+        for(Ship ship : shipList){
+            Request placeRequest = generateValidPlaceRequest(ship);
+            placeShip(placeRequest);
+        }
+    }
+
+    private void removeShip(Ship shipToRemove){
+        Ship[] oldShipList = getShipList();
+        Ship[] newShipList = new Ship[oldShipList.length -  1];
+
+        for(int i = 0, j = 0; i < oldShipList.length; i++){
+            if(oldShipList[i] != shipToRemove){
+                newShipList[j] = oldShipList[i];
+                j++;
+            }
+        }
+        setShipList(newShipList);
+    }
+
+    public void fire(Request hitRequest){
+        playerBoard.hitSection(hitRequest.getRow(), hitRequest.getCol());
+    }
+
```

Comment:
I think it's odd and violates single responsibility principle to have `placeShip`, `placeAllShips`, `removeShip`, and `Fire` within the ComputerPlayer.  The ComputerPlayer seems bloated.  Think of the humanPlayer - or think of requests that come to the controller - you will just have a place request and a hit request.

Reply:
So I've been thinking about this for a while now, and I'm not sure where to move those functions to. Should I remove them altogether, and just let the future Game handle all the Requests?

Reply:
Like, the Computer player generates their Requests, and the Human player will have the logic needed to receive input from the controllers, and then the Game can decide what to do with those?

Reply:
I think a Game class makes a lot of sense